### PR TITLE
Added 'instanceof' binary operation to the API

### PR DESCRIPTION
**Pull request by LaszloLango**
JerryScript-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -1092,6 +1093,7 @@ jerry_get_global_object (void);
 - [jerry_release_value](#jerry_release_value)
 - [jerry_define_own_property](#jerry_define_own_property)
 
+
```
Comment (rerobika) [inline docs/02.API-REFERENCE.md]: This new line is unnecessary.

Comment (LaszloLango) [reply docs/02.API-REFERENCE.md]: IMHO it is necessary. We put two spaces before top level headers (`#`) to make it more readable in raw format. It helps to notice where a new chapter starts. This style is used in the whole MD file.

**Comment by LaszloLango** [comment]
@galpeter good catch, the commit message also wrong. I'll fix it.

File: jerry-core/include/jerryscript-core.h [hunk 2]
```diff
@@ -313,12 +313,13 @@ typedef struct jerry_context_t jerry_context_t;
...
-  JERRY_BIN_OP_LESS_EQUAL,   /**< less or equal relation (<=) */
-  JERRY_BIN_OP_GREATER,      /**< greater relation (>) */
-  JERRY_BIN_OP_GREATER_EQUAL /**< greater or equal relation (>=)*/
+  JERRY_BIN_OP_EQUAL = 0u,    /**< equal comparison (==) */
+  JERRY_BIN_OP_STRICT_EQUAL,  /**< strict equal comparison (===) */
+  JERRY_BIN_OP_LESS,          /**< less relation (<) */
+  JERRY_BIN_OP_LESS_EQUAL,    /**< less or equal relation (<=) */
+  JERRY_BIN_OP_GREATER,       /**< greater relation (>) */
+  JERRY_BIN_OP_GREATER_EQUAL, /**< greater or equal relation (>=)*/
+  JERRY_BIN_OP_INSTANCEOF     /**< instanceof operation */
```
Comment (galpeter) [inline jerry-core/include/jerryscript-core.h]: At other places there is a comma (`,`) after the last value. It is good to follow that as if new operations are added only one line needs changing.

**Pull request merged**